### PR TITLE
Remove force unwrap when encodeType(primaryType for `EIP712Domain` and `EIP712Domain` is missing in typed data #4057

### LINKED ISSUE #4057: WalletConnect 2 eth_signTypedData crash
name: Bug Report
about: It crashed, and rerun got a blank screen.

---

**Describe the bug**

**Steps to reproduce (REQUIRED)**
Steps to reproduce the behavior

1. Connect to WalletConnect 2 test Dapp https://react-app.walletconnect.com/
2. Invoke eth_signTypedData
3. App crashed
4. Rerun

**Expect**
App started.

**Observed**
Blank screen with nothing.

**Screenshots**
image

**Version or build number number**
3.36

### PR DESCRIPTION
**Pull request by oa-s**
Closes #4057

### CODE REVIEW

File: AlphaWallet/WalletConnect/Types/EIP712TypedData.swift [hunk 1]
```diff
@@ -78,8 +78,9 @@ extension EIP712TypedData {
         var depSet = findDependencies(primaryType: primaryType)
         depSet.remove(primaryType)
         let sorted = [primaryType] + Array(depSet).sorted()
-        let encoded = sorted.map { type in
-            let param = types[type]!.map { "\($0.type) \($0.name)" }.joined(separator: ",")
+        let encoded = sorted.compactMap { type in
```
Comment (hboon) [inline AlphaWallet/WalletConnect/Types/EIP712TypedData.swift:81]: Is this because there is an unsupported type that we should support?

Comment (oa-s) [reply AlphaWallet/WalletConnect/Types/EIP712TypedData.swift:81]: i am not quite sure that type is not supported, when we calculate `digest` for typedData we check for `EIP712Domain` that is missing for typed data received from Wallet Connect v2, as i know it should present there

Comment (hboon) [reply AlphaWallet/WalletConnect/Types/EIP712TypedData.swift:81]: Hmm.. OK, can you help leave a comment there before the compactMap to mention the problem we are avoiding? So when we see the code (and the comment) the next time we know?

**Review by hboon — APPROVED** [review]

File: AlphaWallet/WalletConnect/Types/EIP712TypedData.swift (hunk 1 above)
Comment (hboon) [reply AlphaWallet/WalletConnect/Types/EIP712TypedData.swift:81]: Normally I'd rely on `git log -S`, but seems hard to think of searching git log for this :)

**Pull request merged**